microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshot-schema.md

```diff
@@ -0,0 +1,314 @@
+---
+title: Understand the heap snapshot file format
+description: How the heap snapshot files are structured, when exported from the Memory tool
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.date: 9/16/2022
+---
+# Understand the heap snapshot file format
+
+Investigating memory usage in web applications can be hard. The DevTools **Memory** tool allows you to explore all the objects allocated in memory by your web application by taking a heap snapshot. This information is useful for performance investigations because you can find out what objects are consuming the most memory.
+
+However, sometimes you may need to focus on specific parts of the data that the **Memory** tool doesn't show. Fortunately DevTools makes it possible to export the entire memory data as a `heapsnapshot` file which you can then analyze and visualize with your own tools.
```

Comment:
```suggestion
However, sometimes you may need to focus on specific parts of the data that the **Memory** tool doesn't show.  In this case, use DevTools to export the entire set of memory data as a `heapsnapshot` file, which you can then analyze and visualize with your own tools.
```

Reply:
try an alternative to writing "your own tools".  What type of tools would these be, for example?  VS Code / code editor/ text editor?

Reply:
Visual Studio Code isn't necessarily the right example to give here. The format of the data in a `.heapsnapshot` file is too complicated to read as JSON. Later in the doc we mention opening the file in VS Code, but that's just to show people that this is, indeed, just JSON data, and so that they can see real examples of the data documented below.
We're not expecting anyone to actually use VS Code to investigate a memory leak.
The reason for this doc is for people to build their own visualization tools. So I think we should keep the "your own tools" part.